File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int sum = 0;
        for(int i=0; i<k; i++){
            sum = sum + nums[i];
        }
        int maxSum = sum;
        for(int right = k; right<nums.size(); right++){
            int left = right - k;
            sum = sum - nums[left] + nums[right];
            maxSum = max(sum,maxSum);
        }
        return (double)maxSum/k; 
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double maxAverage = Double.NEGATIVE_INFINITY;;
        for(int i=0; i<=nums.length-k; i++){
            int sum = 0;
            for(int j = i; j<i+k; j++){
                sum = sum + nums[j];
            }
            maxAverage = Math.max(maxAverage,(double)sum/k);
        }
        return maxAverage;
    }
}